build/dependency

```diff
@@ -78,49 +132,40 @@ if [ -z "$pkg_prefix" ]; then
 fi
 export pkg_prefix
 cd $pkg_rootdir
-if [ -x build/spec/$1 ]; then
-    exec build/spec/$1  # If the spec is executable, pass control to it
-fi
 # Note that the spec has access to $pkg_rootdir
 . build/spec/$1
-if [ ! -d third_party/cache/$pkg_name ]; then
-    git clone $pkg_clone_flags $pkg_repository third_party/cache/$pkg_name
-    cd third_party/cache/$pkg_name
-    if [ ! -z "$pkg_branch" ]; then
-        git checkout $pkg_branch
-        if [ "`git rev-parse HEAD`" != "$pkg_commit" ]; then
+if [ "$pkg_repository" != "" ]; then
+    rm -rf ./build/third_party/$pkg_name
+    git clone $pkg_clone_flags --single-branch --depth 2                       \
+        --branch $pkg_branch_or_tag $pkg_repository                            \
+          ./build/third_party/$pkg_name
+    cd ./build/third_party/$pkg_name
+    if [ ! -z "$pkg_tip" ]; then
+        if [ "`git rev-parse HEAD`" != "$pkg_tip" ]; then
```

Comment:
The variable pkg_commit has been renamed as pkg_tip to better explain its purpose.